src/ralph/networks/models/networks.py

```diff
@@ -503,11 +513,14 @@ def get_first_free_ip(self):
         for free_ip_as_int in range(min_ip, max_ip + 1):
             if free_ip_as_int not in used_ips:
                 break
-        # TODO: do it better
-        last = free_ip_as_int in used_ips
+        next_free_ip = ipaddress.ip_address(free_ip_as_int)
+        while is_in_dnsaas(next_free_ip) and free_ip_as_int <= max_ip:
+            logger.error('IP {} is already in DNS'.format(next_free_ip))
+            free_ip_as_int += 1
```

Comment:
we should check now if `free_ip_as_int` is not in `used_ips` (just like in line 504/514)